src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -11,9 +12,12 @@ namespace Microsoft.NET.Build.Tasks
 {
     public class PublishAssembliesResolver
     {
+        private static ConcurrentDictionary<PackageInfo, byte> _allResolvedPackages = new ConcurrentDictionary<PackageInfo, byte>();
```

Comment:
This is a bad idea.  We should not be using a static anything here for communication between Tasks.